src/Models/Hooks/useFlattenedModelProperties.ts

```diff
@@ -0,0 +1,117 @@
+
+    /** List of allowed DTDL primitive types to build value properties for */
+    allowedPropertyValueTypes: PropertyValueType[];
+}
+
+const getPropertyModel = (propertyId: string) => {
+    const afterColonSubstring = propertyId.substring(
+        propertyId.lastIndexOf(':') + 1
+    );
+    const modelName = afterColonSubstring.split(';')[0];
+    return modelName;
+};
+
+const groupFlattenedModelProperties = (flattenedModelProperties) => {
+    const groupedProperties = {};
+    Object.keys(flattenedModelProperties).map((key: string) => {
+        flattenedModelProperties[key].map((property) => {
+            const modelName = getPropertyModel(property.key);
+            if (!groupedProperties[modelName]) {
+                groupedProperties[modelName] = [];
+            }
+            groupedProperties[modelName].push(property);
+        });
+    });
+    return groupedProperties;
+};
+
+const filterModelledProperties = (
+    modelDict: ModelDict,
+    allowedPropertyValueTypes: Array<PropertyValueType>
+) => {
+    const modelledProperties = {};
+    const filteredModels: ModelDict = {};
+
+    for (const key in modelDict) {
+        if (modelDict[key].entityKind === 'interface') {
+            filteredModels[key] = modelDict[key];
+        }
+    }
+
+    for (const key in filteredModels) {
+        addInterface(
+            modelledProperties,
+            filteredModels[key] as InterfaceInfo,
+            key,
+            allowedPropertyValueTypes
+        );
+    }
+
+    const flattenedModelPropertiesObject = flattenModelledProperties(
+        modelledProperties['properties']
+    );
+
+    const groupedFlattenedModelProperties = groupFlattenedModelProperties(
+        flattenedModelPropertiesObject
+    );
+
+    return groupedFlattenedModelProperties;
+};
+
+const buildFlattenedModelProperties = async (
+    adapter: IModelledPropertyBuilderAdapter,
+    allowedPropertyValueTypes: PropertyValueType[]
+) => {
+    const modelDict = (await adapter.getAllAdtModels()).getData().parsedModels;
+    const filteredProperties = filterModelledProperties(
+        modelDict,
+        allowedPropertyValueTypes
+    );
+    return filteredProperties;
+};
+
+export const useFlattenedModelProperties = ({
+    adapter,
+    allowedPropertyValueTypes
+}: IUseFlattenedModelPropertiesParams) => {
+    const [isLoading, setIsLoading] = useState(true);
+    const [flattenedModelProperties, setFlattenedModelProperties] = useState(
+        null
+    );
+
+    useEffect(() => {
+        let isMounted = true;
+
+        const updateModelProperties = async () => {
+            const newFlattenedModelProperties = await buildFlattenedModelProperties(
+                adapter,
+                allowedPropertyValueTypes
+            );
+            if (isMounted) {
+                setFlattenedModelProperties(newFlattenedModelProperties);
+                setIsLoading(false);
+            }
+        };
+        updateModelProperties();
```

Comment:
so if you just move the updateModelProperties method out of the useEffect and call it in the useEffect with empty dependency list is going to call it onmount anyhow right? I mean no need to define a isMounted in useeffect and check it with if condition in the updateModelProperties method? maybe I am missing something :)